allauth/socialaccount/providers/microsoft/views.py

```diff
@@ -84,6 +95,34 @@ def complete_login(self, request, app, token, **kwargs):
             )
         )
         extra_data = _check_errors(response)
+
+        if self.fetch_photo:
+
+            photo_metadata_response = (
+                get_adapter()
+                .get_requests_session()
+                .get(
+                    self.photo_metadata_url,
+                    headers=headers,
+                )
+            )
+            if not photo_metadata_response.ok:
```

Comment:
**Question**: How does this behave if the user has no photo set up? Suppose it returns a 404, I don't think the OAuth handshake should hard fail on that...

Reply:
Yes from the [documentation](https://learn.microsoft.com/en-us/graph/api/profilephoto-get?view=graph-rest-1.0&tabs=http#response-for-getting-the-photo) it looks like it would fail if the user has not set up a photo.

I can change to an "If self.fetch_photo is ok:" and put the rest of the logic underneath that if statement.